File: php-advisor-core/src/Payloads/WeatherPayload.php
<?php

namespace StormGeo\AdvisorCore\Payloads;

class WeatherPayload extends BasePayload
{
  /**
   * @var int
   */
	public $localeId;

  /**
   * @var string
   */
	public $latitude;

  /**
   * @var string
   */
	public $longitude;

  /**
   * @var string
   */
  public $stationId;

  /**
   * @var string
   */
	public $startDate;

  /**
   * @var string
   */
  public $endDate;

  /**
   * @var array<string>
   */
	public $variables;

  /**
   * @var int
   */
	public $timezone;

  /**
   * @param array{localeId:int,latitude:string,longitude:string,stationId:string,startDate:string,endDate:string,variables:array<string>,timezone:int} $parameters
   */
  public function __construct($parameters = [])
  {
    parent::__construct(
      ['localeId', 'latitude', 'longitude', 'stationId', 'startDate', 'endDate', 'variables', 'timezone'],
      $parameters
    );
  }

  /**
   * @var array
   */
  public function getQueryParams()
  {
    return (array) $this;
  }
}

File: dotnet-advisor-core/Payloads/WeatherPayload.cs
namespace StormGeo.AdvisorCore.Payloads;

public class WeatherPayload
{
    public int? LocaleId { get; set; }
    public string? Latitude { get; set; }
    public string? Longitude { get; set; }
    public string? StationId { get; set; }
    public string? StartDate { get; set; }
    public string? EndDate { get; set; }
    public string[]? Variables { get; set; }
    public int? Timezone { get; set; }

    public string GetQueryParams()
    {
        return new QueryParameterBuilder()
            .AddLocaleId(LocaleId)
            .AddLatLon(Latitude, Longitude)
            .AddStationId(StationId)
            .AddStartDate(StartDate)
            .AddEndDate(EndDate)
            .AddVariables(Variables)
            .AddTimezone(Timezone)
            .Build();
    }
}
